packages/server/src/api/controllers/automation.ts

```diff
@@ -94,6 +95,13 @@ export async function find(ctx: UserCtx) {
 export async function destroy(ctx: UserCtx<void, DeleteAutomationResponse>) {
   const automationId = ctx.params.id
 
+  const automation = await sdk.automations.get(ctx.params.id)
+  if (
+    automation.definition.trigger.stepId === AutomationTriggerStepId.ROW_ACTION
+  ) {
+    ctx.throw("Row actions cannot be deleted", 403)
```

Comment:
nit: https://www.rfc-editor.org/rfc/rfc9110#status.403 suggests the 403 status is to do with authorisation, I suspect 400 is slightly more accurate here because you're attempting to do something that's invalid regardless of who you are.

Reply:
I changed this 100 times and I did quite a lot of read. I agree that is not a 403, but, at the same time, it's not a 400. The request is not "bad" in terms of [format](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400).

I will change it again to be a 400, as 403 might have more implications and anyways this is something that should never be called

Reply:
Changed to 422, as it seems a good use case for it